jujugui/static/gui/src/app/components/user-profile/entity/entity.js

```diff
@@ -358,6 +468,9 @@ YUI.add('user-profile-entity', function() {
         });
         destroyButton = this._generateDestroyButton(entity);
       } else {
+        if (isCharm) {
```

Comment:
I'd prefer this not to happen every time the component renders. I'm surprised this doesn't error as `this._getMetrics()` causes the state to change which can trigger `render` which, in this case, triggers `this._getMetrics()`. I guess that doesn't happen but it feels like we're at party with a fire pit and at least one person is thinking about playing with the fire.